index.js

```diff
@@ -1,54 +1,32 @@
-const init = (token, { fastUrl } = {}) => {
-  if (typeof chmln !== 'undefined') {
-    console.log(`The global variable \`chmln\` is already defined.\nPlease make sure you're not adding Chameleon to the page other than with this library.`);
-    return;
-  }
 
-  var d = document;
-  var w = window;
-  var t = token,
-  c="chmln",m="identify alias track clear set show on off custom help _data".split(" "),
-  i=d.createElement("script");if(w[c]||(w[c]={}),!w[c].root){w[c].accountToken=t,w[c].location=w.location.href.toString(),w[c].fastUrl=fastUrl||'https://fast.trychameleon.com/',
-      w[c].now=new Date;for(var s=0;s<m.length;s++){!function(){var t=w[c][m[s]+"_a"]=[];w[c][m[s]]=function(){t.push(arguments);};}();}
-      i.src=w[c].fastUrl+"messo/"+t+"/messo.min.js",
-      i.async=!0,d.head.appendChild(i);}
-}
+const noop_available = () => { console.log(`Failed to run because the chameleon.io script was not initialized yet.`);
+return; };
 
-const identify = (id, options) => {
-  if (!window.chmln || !window.chmln.identify) {
-    console.log(`Failed to identify the user [${id}] because the chameleon.io script was not initialized yet.`);
-    return;
-  }
-
-  window.chmln.identify(id, options || {});
-}
-
-const track = api_factory('track');
-const clear = api_factory('clear');
-const set = api_factory('set');
-const show = api_factory('set');
-const on = api_factory('on');
-const off = api_factory('off');
-
-const api_factory = (method_name) => {
-  return function() {
-    if (!chmln || !chmln[method_name]) {
-      console.log(`Failed to call [${method_name}] because the chameleon.io script was not initialized yet.`);
+const chmln_object = {
+  init: (token, { fastUrl } = {}) => {
+    if (typeof chmln !== 'undefined') {
+      console.log(`The global variable \`chmln\` is already defined.\nPlease make sure you're not adding Chameleon to the page other than with this library.`);
       return;
     }
-
-    window.chmln[method_name].apply(this, arguments);
-  }
-}
-
-
-module.exports = {
-  init,
-  identify,
-  track,
-  clear,
-  set,
-  show,
-  on,
-  off
+  
+    var d = document;
+    var w = window;
+    var t = token,
+    c="chmln",m="identify alias track clear set show on off custom help _data".split(" "),
+    i=d.createElement("script");
+    if(w[c]||(w[c]=chmln_object),!w[c].root){w[c].accountToken=t,w[c].location=w.location.href.toString(),w[c].fastUrl=fastUrl||'https://fast.trychameleon.com/',
+        w[c].now=new Date;for(var s=0;s<m.length;s++){!function(){var t=w[c][m[s]+"_a"]=[];w[c][m[s]]=function(){t.push(arguments);};}();}
+        i.src=w[c].fastUrl+"messo/"+t+"/messo.min.js",
+        i.async=!0,d.head.appendChild(i),w[c].init=null;}
+  },
+  identify: noop_available,
+  alias: noop_available,
+  track: noop_available,
+  clear: noop_available,
+  set: noop_available,
+  show: noop_available,
+  on: noop_available,
+  off: noop_available
```

Comment:
@HiAaronCody let's back up one step here -- when you add this JavaScript code snippet to the page (and don't yet identify a user) you get something like this:

<img width="648" alt="Screenshot 2023-09-14 at 8 14 29 AM" src="https://github.com/chamaeleonidae/chmln/assets/401631/7c31708f-5a26-4085-bbd6-3e61e60234c6">

The important part of this is that you end up with "future executed calls" for any of the chmln API functions that you call before the full chmln.js javascript is pulled into the page

before chmln.js is loaded you call

```
chmln.on('foo', () => console.log('on foo'))
chmln.on('track', () => console.log('on track'))
chmln.track('Event name', { opt1: '1' })
```

<img width="648" alt="Screenshot 2023-09-14 at 8 24 17 AM" src="https://github.com/chamaeleonidae/chmln/assets/401631/2aec8217-4dd4-4ad1-9183-2eccbbe238f1">

those wil lthen be called in order when chmln.js is actually loaded.

this now differs from the noop that you have set up here -- tuple me if you want to dive deeper into any of this or if this comment would benefit from a live run though